### PR TITLE
security: add request body size limit to prevent DoS

### PR DESCRIPTION
**Pull request by AlexeySamosadov**
## Summary

Add request body size limit to prevent memory exhaustion DoS attacks.

## Problem

The `readRequestBody` function reads the entire HTTP request body into memory without any size limit:

```go
func readRequestBody(r *http.Request) ([]byte, error) {
    var buf bytes.Buffer
    if _, err := io.Copy(&buf, r.Body); err != nil {  // NO SIZE LIMIT
        return nil, err
    }
    // ...
}
```

An attacker can send arbitrarily large request bodies to `/v1/chat/completions`, causing:
- Unbounded memory allocation
- OOM kill and node crash
- Network-wide DoS if multiple nodes are targeted

## Solution

- Add `MaxRequestBodySize` constant (10 MB)
- Use `http.MaxBytesReader` to enforce the limit
- Requests exceeding the limit are rejected with an error

## Changes

- `decentralized-api/internal/server/public/post_chat_handler.go`
  - Add `MaxRequestBodySize = 10 * 1024 * 1024` constant
  - Wrap `r.Body` with `http.MaxBytesReader` in `readRequestBody`

## Test plan

- [x] Add unit test for normal-sized requests
- [x] Add unit test for oversized requests (should error)
- [x] Add unit test for exactly-at-limit requests
- [x] Add unit test for empty body
- [x] Add test verifying constant value

### CODE REVIEW

File: decentralized-api/internal/server/public/post_chat_handler.go [hunk 1]
```diff
@@ -950,6 +954,9 @@ func readRequest(request *http.Request, transferAddress string) (*ChatRequest, e
 }
 
 func readRequestBody(r *http.Request) ([]byte, error) {
+	// Limit request body size to prevent memory exhaustion attacks
+	r.Body = http.MaxBytesReader(nil, r.Body, MaxRequestBodySize)
```
Comment (patimen) [inline decentralized-api/internal/server/public/post_chat_handler.go:968]: Maybe we should pass in the Writer instead of using nil?

File: decentralized-api/internal/server/public/post_chat_handler_test.go [hunk 2]
```diff
@@ -164,3 +173,57 @@ func TestEmptyButParseableResponsePayload_EnforcedTokensEmptySlice(t *testing.T)
...
+	result, err := readRequestBody(req)
+	require.NoError(t, err)
+	require.Equal(t, body, result)
+}
+
+// TestReadRequestBody_ExceedsMaxSize tests that oversized requests are rejected
+func TestReadRequestBody_ExceedsMaxSize(t *testing.T) {
+	// Create a body larger than MaxRequestBodySize (10 MB)
+	oversizedBody := make([]byte, MaxRequestBodySize+1)
+	for i := range oversizedBody {
```
Comment (patimen) [inline decentralized-api/internal/server/public/post_chat_handler_test.go:191]: It's a nit, but you could use `bytes.Repeat([]byte("a"), MaxRequestBodySize+1)` instead of a for loop.
